### PR TITLE
PAGOPA-1745 adding dynamic mechanism to retrieve station and broker

### PR DESCRIPTION
**Pull request by FedericoRuzzier**
#### List of Changes
Setting up dynamic search for stations and brokers when retrieving the flows.

#### Motivation and Context
Change needed in order not to use one single station and broker (saved in env variables) to retrieve the various flows

#### How Has This Been Tested?
Manually

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -170,8 +183,8 @@
                 <artifactId>maven-compiler-plugin</artifactId>
                 <version>3.8.1</version>
                 <configuration>
-                    <source>${java.version}</source>
-                    <target>${java.version}</target>
+                    <source>16</source>
```
Comment (cap-ang) [inline pom.xml]: why this `${java.version}` is replaced from static value?

File: src/main/java/it/gov/pagopa/reporting/RetrieveFlows.java [hunk 2]
```diff
@@ -45,15 +56,25 @@ public void run(
 
         NodoChiediElencoFlussi nodeClient = this.getNodeClientInstance(logger);
         FlowsService flowsService = this.getFlowsServiceInstance(logger);
+        ApiConfigClient cacheClient = this.getCacheClientInstance();
+        if(cacheContent == null || (cacheContent.getRetrieveDate() != null && cacheContent.getRetrieveDate().isBefore(LocalDateTime.now()))) {
```
Comment (cap-ang) [inline src/main/java/it/gov/pagopa/reporting/RetrieveFlows.java]: `getRetrieveDate` is always before now, no?

File: src/main/java/it/gov/pagopa/reporting/RetrieveFlows.java [hunk 3]
```diff
@@ -94,4 +119,37 @@ public FlowsService getFlowsServiceInstance(Logger logger) {
...
+
+    public Optional<Station> getPAStationIntermediario(String idPa) {
+        List<String> stationPa = getStazioni(idPa);
+        return cacheContent.getStations().stream()
+                .filter(station -> stationPa.contains(station.getStationCode()))
+                .filter(Station::getEnabled)
+                .findFirst();
+    }
+
+    public List<String> getStazioni(String idPa) {
```
Comment (cap-ang) [inline src/main/java/it/gov/pagopa/reporting/RetrieveFlows.java]: can be replaced by `getStations`?

File: src/main/java/it/gov/pagopa/reporting/client/ApiConfigClient.java [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        Type type = new TypeToken<List<CreditorInstitutionStation>>() {}.getType();
+
+        ObjectMapper mapper = new ObjectMapper();
+        CacheResponse cacheResponse = CacheResponse.builder().build();
+        List<CreditorInstitutionStation> creditorInstitutionStationList = new ArrayList<>();
+        List<Station> stationList = new ArrayList<>();
+        try {
+            InputStream resIs = request.execute().getContent();
+            Map<String,Object> responseMap = mapper.readValue(resIs, HashMap.class);
+            Map<String,Object> creditorInstitutionStations = (HashMap) responseMap.get("creditorInstitutionStations");
```
Comment (cap-ang) [inline src/main/java/it/gov/pagopa/reporting/client/ApiConfigClient.java:111]: this cast to `HashMap` is necessary?

File: src/main/java/it/gov/pagopa/reporting/client/ApiConfigClient.java [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+            for (Map.Entry<String, Object> creditorInstitutionStation : creditorInstitutionStations.entrySet()) {
+                creditorInstitutionStationList.add(mapper.readValue(mapper.writeValueAsString(creditorInstitutionStation.getValue()), CreditorInstitutionStation.class));
+            }
+            Map<String,Object> stations = (HashMap) responseMap.get("stations");
```
Comment (cap-ang) [inline src/main/java/it/gov/pagopa/reporting/client/ApiConfigClient.java:115]: ditto

**Review by cap-ang — CHANGES_REQUESTED** [review]
Great work! Check the review please

File: src/main/java/it/gov/pagopa/reporting/client/ApiConfigClient.java (hunk 4 above)
Comment (FedericoRuzzier) [reply src/main/java/it/gov/pagopa/reporting/client/ApiConfigClient.java:111]: Yes, otherwise the mapping of the structure will not work properly

**Review by cap-ang — APPROVED** [review]

**Pull request merged**